modules/apps/27-interchain-accounts/host/keeper/handshake.go

```diff
@@ -47,6 +48,10 @@ func (k Keeper) OnChanOpenTry(
 		return "", err
 	}
 
+	if activeChannelID, found := k.GetOpenActiveChannel(ctx, portID); found {
```

Comment:
Doesn't this effectively restrict host chains to a single interchain account? In `OnChanOpenTry` the `portID` argument is always `"interchain-account"`.

The first handshake to take place will set an active channel in `OnChanOpenConfirm` using the `portID` argument (`"interchain-account"`) as the key, and all subsequent handshakes will then check for an active channel with this portID in `OnChanOpenTry` and fail with an error.

Should we rather use the controller (counterparty) port ID when setting the active channel in `OnChanOpenConfirm`?

Reply:
Nice catch! That's a :bug: . It should be the counterparty port ID

Reply:
You'll note that [ica registration](https://github.com/cosmos/ibc-go/pull/785/files#diff-0af39e5e94b598423a42ea2db6d5611a6c3ab78d653d048bf0a89e9cb2aa7bf1L59) uses the counterparty port ID

Reply:
done. great catch